provider/spark.go

```diff
@@ -811,6 +826,30 @@ func (spark *SparkOfflineStore) sqlTransformation(config TransformationConfig, i
 	return nil
 }
 
+func (spark *SparkOfflineStore) dfTransformation(config TransformationConfig, isUpdate bool) error {
+	transformationDestination := spark.Store.ResourcePath(config.TargetTableID)
+	exists, err := spark.Store.ResourceExists(config.TargetTableID)
+	if err != nil {
+		return err
+	}
+
+	if !isUpdate && exists {
+		return fmt.Errorf("transformation %v already exists at %s", config.TargetTableID, transformationDestination)
+	} else if isUpdate && !exists {
+		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination)
+	}
+
+	sparkArgs, err := spark.getDFArgs(transformationDestination, config.Query, spark.Store.Region(), config.SourceMapping)
+	if err != nil {
+		return fmt.Errorf("error with getting df arguments %v", sparkArgs)
+	}
+
+	if err := spark.Executor.RunSparkJob(sparkArgs); err != nil {
```

Comment:
How does this work atm? Is it creating files with new timestamps for each update?

Reply:
yes, it creates a new timestamp with each update. The design is that we only worry about the latest timestamp.